src/Drive.php

```diff
@@ -577,6 +581,78 @@ public function emptyTrashbin(): bool
         return true;
     }
 
+    /**
+     * Invite a user or group to the drive.
+     * Every recipient will result in an own ShareCreated object in the returned array.
+     *
+     * @param User|Group $recipient
+     * @param SharingRole $role
+     * @param \DateTimeImmutable|null $expiration
+     * @return DriveShare
+     * @throws BadRequestException
+     * @throws ForbiddenException
+     * @throws HttpException
+     * @throws InvalidResponseException
+     * @throws NotFoundException
+     * @throws UnauthorizedException
+     * @throws InternalServerErrorException
+     */
+    public function invite($recipient, SharingRole $role, ?\DateTimeImmutable $expiration = null): DriveShare
+    {
+        $driveItemInviteData = [];
+        $driveItemInviteData['recipients'] = [];
+        $recipientData = [];
+        $recipientData['object_id'] = $recipient->getId();
+        if ($recipient instanceof Group) {
+            $recipientData['at_libre_graph_recipient_type'] = "group";
+        }
+        $driveItemInviteData['recipients'][] = new DriveRecipient($recipientData);
+        $driveItemInviteData['roles'] = [$role->getId()];
+        if ($expiration !== null) {
+            $expirationMutable = \DateTime::createFromImmutable($expiration);
+            $driveItemInviteData['expiration_date_time'] = $expirationMutable;
+        }
+
+        if (array_key_exists('drivesRootApi', $this->connectionConfig)) {
+            $apiInstance = $this->connectionConfig['drivesRootApi'];
```

Comment:
These codes are also inside the resource invite function. I'm sure Sonar Cloud get pissed off with this repetition. https://sonarcloud.io/summary/new_code?id=owncloud_ocis-php-sdk&pullRequest=226